pkg/collector/reconcile/service.go

```diff
@@ -235,6 +235,7 @@ func expectedServices(ctx context.Context, params Params, expected []corev1.Serv
 			updated.ObjectMeta.Labels[k] = v
 		}
 		updated.Spec.Ports = desired.Spec.Ports
+		updated.Spec.Selector = desired.Spec.Selector
```

Comment:
This fixes a bug that wasn't updating the version on services when a collector image would change